theano/gpuarray/basic_ops.py

```diff
@@ -647,14 +647,6 @@ def c_code_cache_version(self):
         return (9,)
 
 
-# Caching GPUAlloc
```

Comment:
You removed the caching, this is good, but to be useful, you should implement the method Op.\__new__() that do the caching automatically based on \__props__. Then we should check all subclass of Op to check its props, as from memory, it will cause some problems.

Can you move that into its own PR and start by implementing the \__new__ method?